lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -124,6 +124,14 @@ def self.run_release_upload(params)
 
         UI.user_error!("Couldn't find build file at path '#{file}'") unless file && File.exist?(file)
 
+        file_ext = Helper::AppcenterHelper.file_extname_full(file)
+        if file_ext == ".app" && File.directory?(file)
+          UI.message("app path is folder, zipping...")
+          app_zipped_path = Actions::ZipAction.run(path: file, output_path: file + ".zip")
```

Comment:
We likely need to delete or move the previous zip file here. Otherwise `zip` will add the new files to the `.zip` archive, and this could result in a larger app.zip containing unwanted files.